src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs

```diff
@@ -583,17 +585,96 @@ public async ValueTask DisposeAsync()
+                        });
+                        var newChangeVector = changeVectorList.SerializeVector();
+
+                        var newDoc = context.ReadObject(document.Data, docId, BlittableJsonDocumentBuilder.UsageMode.ToDisk);
+                        context.DocumentDatabase.DocumentsStorage.Put(context, docId, null, newDoc, document.LastModified.Ticks,
+                            changeVector: newChangeVector,
+                            flags: document.Flags);
+                    }
+
+                    return count;
+                }
+
+                public override TransactionOperationsMerger.IReplayableCommandDto<TransactionOperationsMerger.MergedTransactionCommand> ToDto(JsonOperationContext context)
+                {
+                    return this;
+                }
+
+                public UpdateClusterTransactionIndex ToCommand(DocumentsOperationContext context, DocumentDatabase database)
+                {
+                    return this;
+                }
+            }
+
+
             private async ValueTask SendCommandsAsync(JsonOperationContext context)
             {
                 if (_compareExchangeAddOrUpdateCommands.Count > 0)
                 {
                     var compareExchangeAddOrUpdateCommands = _compareExchangeAddOrUpdateCommands;
-                    await _database.ServerStore.SendToLeaderAsync(new AddOrUpdateCompareExchangeBatchCommand(_compareExchangeAddOrUpdateCommands, context, RaftIdGenerator.DontCareId));
+                    var (index, _) = await _database.ServerStore.SendToLeaderAsync(new AddOrUpdateCompareExchangeBatchCommand(_compareExchangeAddOrUpdateCommands, context, RaftIdGenerator.DontCareId));
                     foreach (var command in compareExchangeAddOrUpdateCommands)
                     {
                         command.Value.Dispose();
                     }
+                    var updateClusterTransactionIndex = new UpdateClusterTransactionIndex
+                    {
+                        Index = index
+                    };
+                    foreach (AddOrUpdateCompareExchangeCommand cmd in _compareExchangeAddOrUpdateCommands)
+                    {
+                        if(ClusterTransactionCommand.IsAtomicGuardKey(cmd.Key, out var docId) == false)
+                            continue;
+                        
+                        updateClusterTransactionIndex.DocumentIds.Add(docId);
+                    }
                     _compareExchangeAddOrUpdateCommands.Clear();
+
+                    if (updateClusterTransactionIndex.DocumentIds.Count > 0)
+                    {
+                        await _database.TxMerger.Enqueue(updateClusterTransactionIndex);
+                        foreach (var docId in updateClusterTransactionIndex.CompareExchangeToRemove)
+                        {
+                            _compareExchangeRemoveCommands.Add(new RemoveCompareExchangeCommand(_database.Name,
```

Comment:
let's use `AddOrUpdateCompareExchangeBatchCommand` it can take multiple `RemoveCommands` ( I know the command name isn't great)

Reply:
That is what is actually happening, we add all those commands, and then send them in a single batch on 

https://github.com/ravendb/ravendb/blob/fe20d9087cfcb71fb824014661cd2c5196b160d5/src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs#L680-L684